src/effects/Skid.tsx

```diff
@@ -0,0 +1,44 @@
+import { Object3D } from 'three'
+import { useRef, useLayoutEffect } from 'react'
+import { useFrame } from '@react-three/fiber'
+import { getState, mutation, useStore } from '../store'
+import type { MutableRefObject } from 'react'
+import type { Euler, InstancedMesh, Vector3 } from 'three'
+import type { Controls } from '../store'
+
+const o = new Object3D()
+
+export function Skid({ count = 500, opacity = 0.5, size = 0.4 }) {
+  const ref = useRef<InstancedMesh>(null!)
+  const { wheels, chassisBody } = useStore((state) => state.raycast)
+
+  let controls: Controls
+  let index = 0
+  useFrame(() => {
+    controls = getState().controls
```

Comment:
I'm not sure but are we now directly getting the state from zustand instead of our useStore()?

Reply:
The useStore got turned into a shallow compare and had all the original useStore methods assigned to it. I just exported the getState method directly so we can skip that step in the future.

Reply:
Got it as long it won't bit back.